mayastor/src/subsys/mod.rs

```diff
@@ -38,3 +42,8 @@ pub(crate) fn register_subsystem() {
         spdk_add_subsystem_depend(Box::into_raw(depend));
     }
 }
+
+/// Add subsystems after the main entrypoint
+pub(crate) fn add_subsystems() {
+    MessageBusSubsystem::add();
```

Comment:
The whole idea of this is that the subsystems run *before* main right? This way, it does not do that.

Reply:
I tried to start it after main but before we start initializing things because there is some kind of issue when connecting the nats client before main:
thread '<unnamed>' panicked at 'assertion failed: c.borrow().is_none()', src/libstd/sys_common/thread_info.rs:40:26

Reply:
Moved the actual bus init into later so now the subsystem starts before main.